### PR TITLE
Add tests for implicit and explicit sharding operations

### PR DESCRIPTION
**Pull request by steventk-g**

### CODE REVIEW

File: test/test_xla_sharding.py [hunk 1]
```diff
@@ -102,6 +102,33 @@ def test_deep_copy(self):
...
+    partition_spec = (0, 1)
+    t1 = torch.tensor([[1, 2, 3, 4, 5, 6, 7, 8]],
+                      dtype=torch.float,
+                      device=xm.xla_device())
+    t2 = torch.tensor([[8, 7, 6, 5, 4, 3, 2, 1]],
+                      dtype=torch.float,
+                      device=xm.xla_device())
+    xs.mark_sharding(t1, self._get_mesh((1, self.n_devices)), partition_spec)
+    t3 = t1 + t2
+    t3expected = [9.0, 9.0, 9.0, 9.0, 9.0, 9.0, 9.0, 9.0]
```
Comment (JackCaoG) [inline test/test_xla_sharding.py:115]: nit `t3_expected `

File: test/test_xla_sharding.py [hunk 2]
```diff
@@ -102,6 +102,33 @@ def test_deep_copy(self):
...
+    t1 = torch.tensor([[1, 2, 3, 4, 5, 6, 7, 8]],
+                      dtype=torch.float,
+                      device=xm.xla_device())
+    t2 = torch.tensor([[8, 7, 6, 5, 4, 3, 2, 1]],
+                      dtype=torch.float,
+                      device=xm.xla_device())
+    xs.mark_sharding(t1, self._get_mesh((1, self.n_devices)), partition_spec)
+    xs.mark_sharding(t2, self._get_mesh((1, self.n_devices)), partition_spec)
+    t3 = t1 + t2
+    t3expected = [9.0, 9.0, 9.0, 9.0, 9.0, 9.0, 9.0, 9.0]
```
Comment (JackCaoG) [inline test/test_xla_sharding.py:129]: ditto

**Comment by yeounoh** [comment]
Let's wait til we resolve b/259442503. Are the tests passing in CI already?

**Comment by yeounoh** [comment]
Closing, @wonjoolee95 will modify and land this on a separate PR.

**Pull request closed without merging**